src/System.Private.CoreLib/shared/System/Numerics/Vector4_Intrinsics.cs

```diff
@@ -340,10 +471,16 @@ public static Vector4 SquareRoot(Vector4 value)
         /// <param name="left">The first vector to compare.</param>
         /// <param name="right">The second vector to compare.</param>
         /// <returns>True if the vectors are not equal; False if they are equal.</returns>
+#if ARM64
         [Intrinsic]
+#endif
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool operator !=(Vector4 left, Vector4 right)
         {
+            if (Sse.IsSupported)
+            {
+                return Sse.MoveMask(Sse.CompareNotEqual(left.AsVector128(), right.AsVector128())) != 0;
```

Comment:
When you just call `!(left == right)` (as in the !Sse.IsSupported path) will the codegen be affected?

Reply:
It shouldn't be. The `==` operator is also aggressively inlined and the `Sse.IsSupported` path will be dropped from there as well, so it should be identical to the code it was previously generating.